### PR TITLE
Implement dynamic channel federation option

### PR DESCRIPTION
**Pull request by stevenhammerton**

### CODE REVIEW

File: lib/irclib/server.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ IrcServer.prototype.getJoinRule = function() {
     return this.config.dynamicChannels.joinRule;
 };
 
+IrcServer.prototype.getFederate = function() {
```
Comment (Kegsay) [inline lib/irclib/server.js]: `shouldFederateRooms` may be more descriptive.

File: spec/integ/dynamic-channels.spec.js [hunk 2]
```diff
@@ -82,6 +82,92 @@ describe("Dynamic channels", function() {
         });
     });
 
+    it("should create non federated room when joining channel and federation is enabled",
```
Comment (Kegsay) [inline spec/integ/dynamic-channels.spec.js]: wha-? Do you mean "should create **a federated room**" rather than "non"?

**Comment by stevenhammerton** [comment]
Kegsay, both comments addressed :)

**Comment by Kegsay** [comment]
LGTM! Thanks!

**Pull request merged**